src/stream.jl

```diff
@@ -286,25 +286,27 @@ end
 
 function Base.seekstart(stream::TranscodingStream)
     mode = stream.state.mode
-    @checkmode (:idle, :read, :write)
-    if mode == :read || mode == :write
+    @checkmode (:idle, :read)
+    if mode == :read
         callstartproc(stream, mode)
-        emptybuffer!(stream.state.buffer1)
-        emptybuffer!(stream.state.buffer2)
+        initbuffer!(stream.state.buffer1)
+        initbuffer!(stream.state.buffer2)
     end
     seekstart(stream.stream)
     return stream
 end
 
 function Base.seekend(stream::TranscodingStream)
     mode = stream.state.mode
-    @checkmode (:idle, :read, :write)
-    if mode == :read || mode == :write
-        callstartproc(stream, mode)
-        emptybuffer!(stream.state.buffer1)
-        emptybuffer!(stream.state.buffer2)
+    buffer1 = stream.state.buffer1
+    if mode == :read
+        while !eof(stream)
+            emptybuffer!(buffer1)
+        end
```

Comment:
I don't think it's a good idea to read the entire file here. Normally, when a user uses `seek`, it's specifically to avoid processing a whole file.
I understand that it may not be possible to seek into a compressed file and get the position of the decompressed stream, but I believe in that case, `seekend` should simply not be implemented, else we risk frustrating users when an operation they believe to be constant time suddenly takes ages.
We can then provide the functionality of reading the whole file under a function with another name.

Reply:
Yes, I agree that it is much better for `Base.seekend(stream::TranscodingStream)` not to be implemented than to silently do the wrong thing. I'm not sure how to go about removing this method though. I created a PR https://github.com/JuliaIO/CodecZlib.jl/pull/68 to remove its use in CodecZlib.jl's tests.

Reply:
I can split this PR into three parts, one for each change, to make reviewing easier.